### PR TITLE
Rework Slicer segmentation plugin

### PR DESCRIPTION
**Pull request by lassoan**
- Reworked  GUI layout
- Fixed volume data caching
- Add options to disable compression of input image to improve speed on fast local network connections
- Show warning before uploading data to remote server
- Save last 10 server addresses
- Added 3 step-by-step tutorials - see here: https://github.com/lassoan/ai-assisted-annotation-client/tree/slicer-ui-rework/slicer-plugin
- Removed support for Slicer-4.10.2, as SimpleITK bundled in this Slicer version did not seem to be compatible with latest version of AIAA client library (this is not a big issue, as we'll release new Slicer Stable Release soon and until then the Preview Releases can be used)

### CODE REVIEW

**Comment by SachidanandAlle** [comment]
Merged.. but i see a small issue.. when i do auto-segment (say for spleen)
it creates new segment instead of editing/modifying the selected one....

Screenshot from 2019-12-16 11-12-17

**Comment by lassoan** [comment]
Auto-segment may create several segments from scratch, without any user inputs. It is for the user's convenience that he does not have to pre-create any segments but all the segments are created automatically.

Annotation is different because it requires user input, which the user may want to tune and then re-run the annotation. Also, most annotation models create only a single segment.

You can find example workflow descriptions here: https://github.com/NVIDIA/ai-assisted-annotation-client/tree/master/slicer-plugin#tutorials-and-examples. Let me know if you have suggestions for improvements.

File: slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py [hunk 1]
```diff
@@ -56,268 +66,240 @@ def icon(self, name='SegmentEditorEffect.png'):
...
-            if addedSegments == 0:
-                segment.SetColor(color)
-                segment.SetName(label)
-
-                self.scriptedEffect.parameterSetNode().SetSelectedSegmentID(segmentId)
-                logging.debug('Extreme Points: {}'.format(extreme_points))
-                if extreme_points is not None:
-                    segment.SetTag("DExtr3DExtremePoints", json.dumps(extreme_points))
-
+            if i==0 and overwriteCurrentSegment and currentSegment:
```
Comment (SachidanandAlle) [inline slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py]: i guess some issue with this condition... not modifying the current selected segment..

Comment (lassoan) [reply slicer-plugin/NvidiaAIAA/SegmentEditorNvidiaAIAALib/SegmentEditorEffect.py]: The current behavior is intentional. See https://github.com/NVIDIA/ai-assisted-annotation-client/pull/47#issuecomment-566203912

**Comment by SachidanandAlle** [comment]
> Auto-segment may create several segments from scratch, without any user inputs. It is for the user's convenience that he does not have to pre-create any segments but all the segments are created automatically.
> 
> Annotation is different because it requires user input, which the user may want to tune and then re-run the annotation. Also, most annotation models create only a single segment.
> 
> You can find example workflow descriptions here: https://github.com/NVIDIA/ai-assisted-annotation-client/tree/master/slicer-plugin#tutorials-and-examples. Let me know if you have suggestions for improvements.

i see now.. we have changed the first behavior here to make easier for user..  :)

after loading the image, user need not to create any label for auto-segmentation... user can directly select the model and run auto-segmentation.. 👍

**Pull request merged**